core/controllers/admin.py

```diff
@@ -225,6 +227,12 @@ class AdminHandler(
                 },
                 'default_value': None
             },
+            'blog_post_title': {
+                'schema': {
+                    'type': 'basestring'
+                },
+                'default_value': None
```

Comment:
Couldn't we just require this in the frontend, and not allow the action to happen unless it is provided? 

Also, can we modify this schema to catch the "not provided" case? Then we don't need the exception below since the schema validation here should take care of any errors. (Just make sure we have a unit test for that case and confirm that it raises a 400 error.)

IIRC there might also be a "choices" option that you can use to narrow down the allowed strings to only certain ones.

Reply:
Done.

Reply:
> Also, can we modify this schema to catch the "not provided" case?

`blog_post_title` should be the optional field, since it is not required in other methods of AdminHandler. We will need the exception as we need to verify the presence of blog_post_title only if action is `generate_dummy_blog_post`.

> IIRC there might also be a "choices" option that you can use to narrow down the allowed strings to only certain ones.
Yeah will use that.